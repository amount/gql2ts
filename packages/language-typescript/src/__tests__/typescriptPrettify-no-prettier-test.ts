jest.mock('prettier', () => { throw new Error('Prettier not installed!'); });

import prettify, { PRETTY_PRINT_WARNING } from '../typescriptPrettify';

const tsString: string = `export interface ITestInterface { a:string;b:string|number;c:string|number|boolean; }`;

describe('typescript prettify', () => {
  it ('succeeds if prettier is missing', () => {
    const spy: jest.SpyInstance<(message: any, ...args: any[]) => void> = jest.spyOn(console, 'warn');
    expect(prettify(tsString)).toMatchSnapshot();
    prettify(tsString);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(PRETTY_PRINT_WARNING);
  });
});
