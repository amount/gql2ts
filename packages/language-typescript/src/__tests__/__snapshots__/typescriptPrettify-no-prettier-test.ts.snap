// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript prettify succeeds if prettier is missing 1`] = `"export interface ITestInterface { a:string;b:string|number;c:string|number|boolean; }"`;
